vconfig/p2pconfig/p2p_cli_config.go

```diff
@@ -0,0 +1,106 @@
+package p2pconfig
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/libsv/bitcoin-hc/version"
+)
+
+// runServiceCommand is only set to a real function on Windows.  It is used
+// to parse and execute service commands specified via the -s flag.
+var runServiceCommand func(string) error
+
+// serviceOptions defines the configuration options for the daemon as a service on
+// Windows.
+type serviceOptions struct {
+	ServiceCommand string `short:"s" long:"service" description:"Service command {install, remove, start, stop}"`
+}
+
+// newConfigParser returns a new command line flags parser.
+func newConfigParser(cfg *Config, so *serviceOptions, options flags.Options) *flags.Parser {
+	parser := flags.NewParser(cfg, options)
+	if runtime.GOOS == "windows" {
+		_, err := parser.AddGroup("Service Options", "Service Options", so)
+		if err != nil {
+			cfg.Logger.Error(err)
+		}
+	}
+	return parser
+}
+
+// ParseFlags returns Config from flags.
+func ParseFlags(customWorkingDirectory string) (*Config, error) {
```

Comment:
If I understand correctly, `customWorkingDirectory` means a directory where a default config file for p2p is. Maybe instead of putting it as function parameter here and in `DefaultP2PConfig()` function, we could use it as a normal flag and/or env variable just as other config options?

Because now, we have to put an empty string into the parameter for default values and it feels weird.

Also, the name doesn't suggest that it's a directory for storing config file, so maybe something like `configFileDirectory` or `customConfigDirectory` would be better? 

WDYT?

Reply:
This solution with `customWorkingDirectory` is taken from BUX. It's necessary mostly for tests, not for regular usage. If you want to use a custom you can specify by --configfile or -C flag.
Regarding the name, you're right. Not exactly a working directory, it is more rootDirectory

Reply:
Mechanism was changed. customWorkingDirectory can be forgotten like a bad dream.

Reply:
Nice!